doc/performance_blog_post/performance_blog_post.md

```diff
@@ -0,0 +1,120 @@
+
+This dataset consists of pseudo-random generated strings. The different test cases are built as show in the following table:
+
+| Test Case                          | String size in byte | Rows      | Files | Columns |
+|------------------------------------|---------------------|-----------|-------|---------|
+| testLoadManyParquetRows            | 100                 | 1000000   | 10    | 1       |
+| testLoadLargeParquetRows           | 1,000,000           | 100       | 10    | 1       |
+| testLoadManyParquetColumns         | 1000                | 1000      | 10    | 100     |
+| testLoadManyParquetRowsFromOneFile | 1,000               | 1,000,000 | 1     | 1       |
+
+You can observe that the product of all columns is always 1,000,000,000 so 1 GB.
+
+The last two tests use exemplary sales data consisting of 7 decimal columns with 3,621,459,710 rows. These two tests split the the sales data into 8 (`testLoadFewBigSalesParquetFiles`) and 200 (`testLoadSalesParquetFiles`) files respectively.
+
+![Performance test results of Parquet file loading 1](slowParquetTestResults1.png)
+![Performance test results of Parquet file loading 2](slowParquetTestResults2.png)
+
+From the charts, it becomes clear that the Virtual Schema at that point has an issue with loading data from a few big files. When loading from more files, it loads the same amount of data a lot quicker.
+
+Starting with this analysis, we discussed with our product management and decided to invest in performance improvements for loading big Parquet files. More precisely we set the goal to make improvements so that the `testLoadFewBigSalesParquetFiles` also runs in less than 250s.
+
+## Improving the Performance
+
+Next, we analyzed the source code for possible reasons for the slow reading of big Parquet files. We identified the following issues:
+
+* Unequal distribution
+* Less parallelization for a few Parquet files
+
+The query processing of the Files Virtual Schemas consists of two phases: A planning phase and an execution phase. In the planning phase, the adapter analyzes the query and starts multiple workers, using [User Defined Functions (UDFs)](https://docs.exasol.com/database_concepts/udf_scripts.htm), that then run the execution phase. Thus, Virtual Schema parallelizes the execution of external data import over multiple workers. For technical reasons, the workers can not communicate at the moment.
+
+### Distributing the File
+
+Until now this parallelization was realized using hash binning. So each worker received a range and then loaded files where the hashs value of the file-name fell into that range. That approach is great for distributing lots of files since it does not require listing files in advance. Doing so could cause memory overflow when distributing for example 1,000,000 files with long names.
```

Comment:
hash value, hash values, hash's values?

Reply:
```suggestion
Until now this parallelization was realized using hash binning. With this approach, each worker receives a range and then loads files where the hash value of the file-name fells into that range. That approach is great for distributing lots of files since it does not require listing files in advance. Doing so could cause memory overflow when distributing for example 1,000,000 files with long names.
```